src/main/java/com/minecolonies/colony/permissions/Permissions.java

```diff
@@ -427,7 +427,7 @@ public boolean addPlayer(String player, Rank rank)
         GameProfile gameprofile = MinecraftServer.getServer().getPlayerProfileCache().getGameProfileForUsername(player);
 
         //Check if the player already exists so that their rank isn't overridden
-        return gameprofile != null && players.get(gameprofile.getId()) != null && addPlayer(gameprofile, rank);
+        return gameprofile != null && players.get(gameprofile.getId()) == null && addPlayer(gameprofile, rank);
```

Comment:
I believe I already fixed this in a different branch by changing it to !players.contains(gameprofile.getId()), So you should either revert your changes and merge the latest develop into your branch, or change it to that so there aren't conflicts.